packages/react/src/jsx.ts

```diff
@@ -0,0 +1,36 @@
+import { ComponentDataState, JSONValue } from './template'
+
+export function jsx(
+  componentFactory: (props: any) => React.ReactElement,
+  { children, ...passedProps }: Record<string, JSONValue>,
+  key: string
+): ComponentDataState {
+  const { props, type } = componentFactory(passedProps)
+  if (typeof type !== 'string') {
+    throw new Error('Invalid component.')
+  }
+  const serialisedProps = Object.fromEntries<typeof props>(
+    Object.entries(props).map(([key, value]) => {
+      // tbd: create a global handler registry and register this callback
+      // make sure to clean-up when no longer needed
+      if (typeof value === 'function') {
+        const uuid = 'generate-uuid-for-handler'
+        return [
+          key,
+          {
+            $: 'event',
+            action: uuid,
```

Comment:
I know we discussed about using paths here, but we don't know the path here as we're not rendering on the server. Theoretically, we could compute path on the server and pass complete pre-rendered tree to the client, but it would mean JSX is now required and outputting a custom JSON (different language, or manually) becomes a challenge. It would also be tricky when refs are used to cross-reference a component or something or there is layout mutation on the client and path is no longer valid.

Reply:
UUID of some sort should be good enough to register and then, de-register (after state update, check if UUID is still present. If not, perform a clean-up)